node/builder_chain.go

```diff
@@ -280,14 +277,17 @@ func ConfigFullNode(c interface{}) Option {
 			),
 		),
 
-		// enable message index for full node when configured by the user, otherwise use dummy.
-		If(cfg.Index.EnableMsgIndex, Override(new(index.MsgIndex), modules.MsgIndex)),
-		If(!cfg.Index.EnableMsgIndex, Override(new(index.MsgIndex), modules.DummyMsgIndex)),
-
 		// enable fault reporter when configured by the user
 		If(cfg.FaultReporter.EnableConsensusFaultReporter,
 			Override(ConsensusReporterKey, modules.RunConsensusFaultReporter(cfg.FaultReporter)),
 		),
+
+		ApplyIf(isFullNode,
```

Comment:
any else case needed here? (`!isFullNode`) Are there things that will need this?

I think we have lite node cases covered in itests so hopefully it gets caught there

Reply:
I don't think so because itests are green and both the `StateManager` and `EthModule` that use the `ChainIndexer` can handle a nil `ChainIndexer` without blowing up.